reference/docs-conceptual/learn/shell/using-predictors.md

```diff
@@ -0,0 +1,149 @@
+**PredictionSource** parameter. The **PredictionSource** can be set to:
+
+- `None`
+- `History`
+- `Plugin`
+- `HistoryAndPlugin`
+
+### Setting the prediction color
+
+By default, predictions appear in light grey text on the same line the user is typing. To support
+accessibility needs, you can customize the prediction color. Colors are defined using ANSI escape
+sequences. You can use `$PSStyle` to compose ANSI escape sequences.
+
+```powershell
+Set-PSReadLineOption -Colors @{ InlinePrediction = $PSStyle.Background.Blue }
+```
+
+Or you can create your own. The default light-grey prediction text color can be restored using the
+following ANSI escape sequence.
+
+```powershell
+Set-PSReadLineOption -Colors @{ InlinePrediction = "`e[38;5;238m" }
+```
+
+For more information about setting prediction color and other PSReadLine settings, see
+[Set-PSReadLineOption][Set-PSReadLineOption].
+
+### Changing keybindings
+
+PSReadLine contains functions to navigate and accept predictions. For example:
+
+- `AcceptSuggestion` - Accept the current inline suggestion
+- `AcceptNextSuggestionWord` - Accept the next word of the inline suggestion
+- `AcceptSuggestion` is built within `ForwardChar`, which is bound to
+  <kdb>RightArrow</kbd> by default
+- `AcceptNextSuggestionWord` is built within the function `ForwardWord`, which can be bound to
+  <kbd>Ctrl</kbd>+<kbd>f</kbd>
+
+You can use the `Set-PSReadLineKeyHandler` cmdlet to change key bindings.
+
+```powershell
+Set-PSReadLineKeyHandler -Chord "Ctrl+f" -Function ForwardWord
+```
+
+With this binding, pressing <kbd>Ctrl</kbd>+<kbd>f</kbd> accepts the next word of an inline
+suggestion when the cursor is at the end of current editing line. You can bind other keys to
+`AcceptSuggestion` and `AcceptNextSuggestionWord` for similar functionalities. For example, you may
+want to make <kdb>RightArrow</kbd> accept the next word of the inline suggestion, instead of the
+whole suggestion line.
+
+```powershell
+Set-PSReadLineKeyHandler -Chord "RightArrow" -Function ForwardWord
+```
+
+## Using other predictor plug-ins
+
+The **Az.Tools.Predictor** module was the first plug-in for Predictive IntelliSense. It uses Machine
+Learning to predict what Azure PowerShell command you want to run and the parameters you want to
+use. For more information and installation instructions, see
+[Announcing General Availability of Az.Tools.Predictor][azpredictor].
+
+The **CompletionPredictor** module adds an IntelliSense experience for anything that can be
+tab-completed in PowerShell. With **PSReadLine** set to `InlineView`, you get the normal tab
+completion experience. When you switch to `ListView`, you get the IntelliSense experience.
+
+![PowerShell IntelliSense using the CompletionPredictor](media/using-predictors/completion-predictor.gif)
+
+You can install the [CompletionPredictor][CompletionPredictor] module from the PowerShell Gallery.
+
```

Comment:
We should add a section that links to the creating your own predictor docs.

Reply:
done